qadence_protocols/measurements/tomography.py

```diff
@@ -6,89 +6,110 @@
 from qadence.blocks.utils import unroll_block_with_scaling
 from torch import Tensor
 
+from qadence_protocols.measurements.abstract import MeasurementManager
 from qadence_protocols.measurements.utils_tomography import (
     convert_samples_to_pauli_expectation,
     iterate_pauli_decomposition,
 )
-
-
-def compute_measurements(
-    model: QuantumModel,
-    observables: list[AbstractBlock],
-    options: dict,
-    param_values: dict[str, Tensor] = dict(),
-    state: Tensor | None = None,
-) -> list:
-    """Obtain measurements by sampling via rotated circuits in Z-basis.
-
-    Args:
-        model (QuantumModel): Model to evaluate.
-        observables (list[ConvertedObservable]): a list of observables
-            to estimate the expectation values from.
-        options (dict): Tomography options.
-        Must be a dict with a 'n_shots' key and integer value.
-        param_values (dict[str, Tensor], optional): Parameter values. Defaults to dict().
-        state (Tensor, optional): Input state. Defaults to dict().
-
-    Raises:
-        TypeError: If observables are not given as list
-        KeyError: If a 'n_shots' kwarg of type 'int' is missing in options.
-
-    Returns:
-        list: List of samples per observable.
-    """
-
-    n_shots = options["n_shots"]
-    circuit = model._circuit.original
-
-    samples = []
-    for observable in observables:
-        pauli_decomposition = unroll_block_with_scaling(observable)
-        samples.append(
-            iterate_pauli_decomposition(
-                circuit=circuit,
-                param_values=model.embedding_fn(model._params, param_values),
-                pauli_decomposition=pauli_decomposition,
-                n_shots=n_shots,
-                state=state,
-                backend=model.backend,
-                noise=model._noise,
+from qadence_protocols.types import MeasurementData
+
+
+class Tomography(MeasurementManager):
+    """The abstract class that defines the interface for the managing measurements."""
+
+    def __init__(self, measurement_data: MeasurementData = None, options: dict = dict()):
+        self.measurement_data = measurement_data
+        self.options = options
+
+    def verify_options(self) -> dict:
+        """Verify options contain `n_shots`.
+
+        Raises:
+            KeyError: If `n_shots` absent from options.
+
+        Returns:
+            dict: Options if correct.
+        """
+        n_shots = self.options.get("n_shots")
+        if n_shots is None:
+            raise KeyError("Tomography protocol requires a 'n_shots' kwarg of type 'int').")
+        return self.options
+
+    def reconstruct_state(self, snapshots: Tensor) -> Tensor:
+        raise NotImplementedError
+
+    def get_snapshots(
+        self,
+        model: QuantumModel,
+        param_values: dict[str, Tensor] = dict(),
+        state: Tensor | None = None,
+    ) -> Tensor:
+        raise NotImplementedError
```

Comment:
I think it would be better not to define them here if possible. Maybe create `ShadowManager` subclass that would hold these abstract methods ?

Reply:
Ok, let me indeed separate them.

Reply:
Done

Reply:
Then you can remove them ?